File: jdollarx/src/main/java/info/testtools/jdollarx/ElementProperty.java
package info.testtools.jdollarx;


public interface ElementProperty {
    String toXpath() ;

    default ElementProperty or(ElementProperty p) {
        return new ElementProperties.Or(this, p);
    }

    default ElementProperty and(ElementProperty prop) {
        return new ElementProperties.And(this, prop);
    }

    default ElementProperty andNot(ElementProperty prop) { return and(new ElementProperties.Not(prop)); }
}

File: dollarx/src/main/java/info/dollarx/ElementProperty.scala
package info.dollarx


trait ElementProperty {

    def toXpath: String

    def or(p: ElementProperty): ElementProperty = {
      return new ElementPropertiesHelper.Or(this, p)
    }

    def and(prop: ElementProperty): ElementProperty = {
      return new ElementPropertiesHelper.And(this, prop)
    }

    def andNot(prop: ElementProperty): ElementProperty = {
      return and(new ElementPropertiesHelper.Not(prop))
    }
}
